integration-tests/src/test/groovy/com/okta/sdk/tests/it/IdpIT.groovy

```diff
@@ -0,0 +1,288 @@
+            .setResponseSignatureScope(ProtocolAlgorithmTypeSignature.ScopeEnum.ANY)
+            .setAcsEndpointBinding(ProtocolEndpoint.BindingEnum.POST)
+            .setAcsEndpointType(ProtocolEndpoint.TypeEnum.INSTANCE)
+            .setAuthorizationEndpointBinding(ProtocolEndpoint.BindingEnum.REDIRECT)
+            .setAuthorizationEndpointUrl("https://idp.example.com/authorize")
+            .setTokenEndpointBinding(ProtocolEndpoint.BindingEnum.POST)
+            .setTokenEndpointUrl("https://idp.example.com/token")
+            .setUserInfoEndpointBinding(ProtocolEndpoint.BindingEnum.REDIRECT)
+            .setUserInfoEndpointUrl("https://idp.example.com/userinfo")
+            .setJwksEndpointBinding(ProtocolEndpoint.BindingEnum.REDIRECT)
+            .setJwksEndpointUrl("https://idp.example.com/keys")
+            .setScopes(["openid", "profile", "email"])
+            .setProtocolType(Protocol.TypeEnum.OIDC)
+            .setClientId("your-client-id")
+            .setClientSecret("your-client-secret")
+            .setBaseUrl("https://idp.example.com")
+            .setMaxClockSkew(120000)
+            .setSubjectTemplate("idpuser.email")
+            .setPolicySubjectMatchType(PolicySubjectMatchType.USERNAME)
+            .buildAndCreate(client)
+        registerForCleanup(createdIdp)
+
+        // list linked idp users
+        assertThat(createdIdp.listUsers(), iterableWithSize(0))
+
+        // link user
+        IdentityProviderApplicationUser idpAppUser = createdIdp.linkUser(createdUser.getId(),
+            client.instantiate(UserIdentityProviderLinkRequest)
+                .setExternalId("externalId"))
+
+        assertThat(createdIdp.listUsers(), iterableWithSize(1))
+        assertPresent(createdIdp.listUsers(), idpAppUser)
+
+        // unlink user
+        createdIdp.unlinkUser(createdUser.getId())
+
+        // list linked idp users
+        assertThat(createdIdp.listUsers(), iterableWithSize(0))
+
+        // list social auth tokens
+        SocialAuthTokenList socialAuthTokenList = createdIdp.listSocialAuthTokens(createdUser.getId())
+        assertThat(socialAuthTokenList, iterableWithSize(0))
+
+        // deactivate
+        createdIdp.deactivate()
+
+        // delete
+        createdIdp.delete()
+    }
+
+    @Test
+    void googleIdpTest() {
+
+        // create user
+        def email = "joe.coder+${uniqueTestName}@example.com"
+        User createdUser = UserBuilder.instance()
+            .setEmail(email)
+            .setFirstName("Joe")
+            .setLastName("Code")
+            .setPassword("Password1".toCharArray())
+            .buildAndCreate(client)
+        registerForCleanup(createdUser)
+
+        // create google idp
+        IdentityProvider createdIdp = GoogleIdentityProviderBuilder.instance()
+            .setName("Mock Google IdP")
+            .setScopes(["openid", "profile", "email"])
+            .setProtocolType(Protocol.TypeEnum.OAUTH2)
```

Comment:
Is there another valid protocol type? for this?

Reply:
Per https://developer.okta.com/docs/reference/api/idps/#protocol-object, `Google` IdP should use `OIDC` type. I'll correct this statement here. Not sure how this IT passes with the present `OAUTH2` setting.